### DISCUSSION TITLE
Failed to save: uploadChunk failed: Cache service responded with 500

### DISCUSSION THREAD

**Discussion by itss050** [community/community#160604] [Actions]
Looking for help on how to debug cache failing to save.

I have two separate instances of GHES running.  One works as expected however the other fails saving the cache.  I am unable to find anything in the logs and the workflow output isn't helpful.

`
##[debug]File Size: 3395
##[debug]Reserving Cache
##[debug]Resource Url: https://mydomain.com/_services/artifactcache/U0Hz6J5v04j8ziLeRZlN5c5BcCqyGoyz6VhmnrktxpMkvCN6JH/_apis/artifactcache/caches
##[debug]Saving Cache (ID: 108)
##[debug]Upload cache
##[debug]Resource Url: https://mydomain.com/_services/artifactcache/U0Hz6J5v04j8ziLeRZlN5c5BcCqyGoyz6VhmnrktxpMkvCN6JH/_apis/artifactcache/caches/108
##[debug]Upload concurrency: 4
##[debug]Upload chunk size: 33554432
##[debug]Awaiting all uploads
##[debug]Uploading chunk of size 3395 bytes at offset 0 with content range: bytes 0-3394/*
##[debug]uploadChunk (start: 0, end: 3394) - Attempt 1 of 2 failed with error: Cache service responded with 500
##[debug]uploadChunk (start: 0, end: 3394) - Error is not retryable
Warning: Failed to save: uploadChunk (start: 0, end: 3394) failed: Cache service responded with 500
Cache saved with key: Linux-primes-v5
##[debug]Node Action run completed with exit code 0
##[debug]Finishing: Save Primes
`

The UI shows conflicting info on a cache existing.  Calling APIs directory returns no results in the cache
![image](https://github.com/user-attachments/assets/780619fc-9d2b-4956-aba8-ac4ae1015b35)

**Comment by Tehmeena0** [comment]
It sounds like a frustrating issue, especially with the logs not providing much clarity. Since one of your GHES instances works fine while the other doesn't, it might be worth checking if there are any differences in the configuration or network settings between the two. Also, ensure that both instances are running the same versions of GHES and that there's no proxy or firewall interference on the failing instance.

If possible, try increasing the logging level to see if it provides more detailed information. Sometimes, even a small discrepancy can cause issues like this. If the problem persists, you might want to reach out to GitHub Support for further investigation.

**Comment by DeveloperVaibhav1** [comment]
A **500 error from the cache service** usually indicates a server-side problem rather than an issue with the workflow itself. Since one GHES instance works and the other does not, it may be related to configuration differences between the two environments.

Some things you might want to check:

* Ensure both GHES instances are running the **same version** and that the Actions cache service is properly configured.
* Verify there are **no proxy, firewall, or network restrictions** blocking the artifact cache endpoints.
* Check available **storage and disk limits**, since cache uploads can fail if the backend storage is not available.
* If possible, enable more detailed logging on the failing instance to see if the cache service returns additional errors.

If the issue continues, it may be best to **contact GitHub Enterprise Support**, since cache service errors (500) are often related to the server configuration rather than the workflow.

**Comment by DeveloperVaibhav1** [comment]
A `500` error from the cache service usually indicates a server-side issue rather than a problem with the workflow itself. Since one GHES instance works and the other fails, it may be related to differences in configuration between the two environments.

A few things you might want to check:

* Make sure both GHES instances are running the **same version**.
* Verify there are **no proxy or firewall restrictions** blocking the artifact cache endpoints.
* Check **available storage/disk limits**, since cache uploads can fail if the backend storage is unavailable.
* Enable more **detailed logging** on the failing instance to see if the cache service returns additional information.

If the problem persists, it may be worth contacting **GitHub Enterprise Support**, as `500` errors are often related to the cache service configuration on the server.
